### PR TITLE
Added application-dev.yml and injected default properties

### PR DESCRIPTION
**Pull request by abhishekaypurohit**
Added application-dev.yml and injected default properties

`BUILD SUCCESSFUL in 4m 33s
348 actionable tasks: 47 executed, 301 up-to-date`

@pdarvasi @lacikaaa @keyki @handavid

### CODE REVIEW

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java [hunk 1]
```diff
@@ -46,15 +46,26 @@
     @Value("#{'${cb.enabled.linux.types}'.split(',')}")
     private List<String> enabledLinuxTypes;
 
+    @Value("${cb.image.catalog.url}")
```
Comment (pdarvasi) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java]: This is already there in https://github.com/hortonworks/cloudbreak/blob/master/freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageService.java#L63-L67

File: freeipa/src/main/resources/application-dev.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+cb.image.catalog.url: https://cloudbreak-imagecatalog.s3.amazonaws.com/freeipa-image-catalog.json
```
Comment (pdarvasi) [inline freeipa/src/main/resources/application-dev.yml]: This is already there in: https://github.com/hortonworks/cloudbreak/blob/master/freeipa/src/main/resources/application.yml#L512

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java [hunk 3]
```diff
@@ -46,15 +46,26 @@
...
     @Inject
     private ObjectMapper objectMapper;
 
     @Cacheable(cacheNames = "imageCatalogCache", key = "#catalogUrl")
     public ImageCatalog getImageCatalog(String catalogUrl)  {
         ImageCatalog catalog;
-        if (catalogUrl == null) {
-            LOGGER.info("No image catalog was defined!");
-            return null;
+        if (StringUtils.isEmpty(catalogUrl)) {
```
Comment (pdarvasi) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java]: This case is handled in ImageService

**Comment by pdarvasi** [comment]
May I ask what was your issue? I've just tried out a create request with empty "Image" section and it found the image successfully.

**Comment by pdarvasi** [comment]
I've seen the issue in chat, the easy way to get it working is by omitting "catalog" section in the request.

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java (hunk 1 above)
Comment (handavid) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java]: hmm. I didn't intend to change anything related to the Image Catalog work. let me see if the diff is picking something up something due to rebasing issues.

**Pull request closed without merging**